src/UndoController.js

```diff
@@ -0,0 +1,72 @@
+var diff = require('virtual-dom/diff');
+var patch = require('virtual-dom/patch');
+var virtualize = require('vdom-virtualize');
+var uuid = require('uuid');
+
+module.exports = class UndoController {
+
+  constructor(scribe, attrs){
+    //remove the default undo command as we will be replacing it
+    scribe.commands.undo.execute = function(){};
```

Comment:
To me this plugin IS a monkey patch as Scribe has no idea of 'data' which makes this kind of operation almost imposible. Ultimately once we have decided to fix Scribe then this plugin will be redundant.